api/base/serializers.py

```diff
@@ -106,103 +114,65 @@ def to_internal_value(self, data):
     """
 
-    def __init__(self, field, **kwargs):
-        super(HideIfRegistration, self).__init__(**kwargs)
-        self.field = field
-        self.source = field.source
-        self.required = field.required
-        self.read_only = field.read_only
-
-    def get_attribute(self, instance):
-        if instance.is_registration:
-            if isinstance(self.field, RelationshipField):
-                raise SkipField
-            else:
-                return None
-        return self.field.get_attribute(instance)
+    def __init__(self, field, min_version, max_version, **kwargs):
+        super(ShowIfVersion, self).__init__(field, **kwargs)
+        self.min_version = min_version
+        self.max_version = max_version
+        self.help_text = 'This field is deprecated as of version {}'.format(self.max_version) or kwargs.get('help_text')
 
-    def bind(self, field_name, parent):
-        super(HideIfRegistration, self).bind(field_name, parent)
-        self.field.bind(field_name, self)
+    def should_hide(self, instance):
+        request = self.context.get('request')
+        return request and utils.is_deprecated(request.version, self.min_version, self.max_version)
 
-    def to_internal_value(self, data):
-        return self.field.to_internal_value(data)
 
-    def to_representation(self, value):
-        if getattr(self.field.root, 'child', None):
-            self.field.parent = self.field.root.child
-        else:
-            self.field.parent = self.field.root
-        return self.field.to_representation(value)
+class ShowIfCurrentUser(ConditionalField):
 
-    def to_esi_representation(self, value, envelope='data'):
-        if getattr(self.field.root, 'child', None):
-            self.field.parent = self.field.root.child
-        else:
-            self.field.parent = self.field.root
-        return self.field.to_esi_representation(value, envelope)
+    def should_show(self, instance):
+        request = self.context.get('request')
+        return request and request.user == instance
 
 
-class HideIfDisabled(ser.Field):
+class HideIfRegistration(ConditionalField):
     """
-    If the user is disabled, returns None for attribute fields, or skips
-    if a RelationshipField.
+    If node is a registration, this field will return None.
     """
 
-    def __init__(self, field, **kwargs):
-        super(HideIfDisabled, self).__init__(**kwargs)
-        self.field = field
-        self.source = field.source
-        self.required = field.required
-        self.read_only = field.read_only
+    def should_hide(self, instance):
+        return instance.is_registration
 
-    def get_attribute(self, instance):
-        if instance.is_disabled:
-            if isinstance(self.field, RelationshipField):
-                raise SkipField
-            else:
-                return None
-        return self.field.get_attribute(instance)
+    def should_be_none(self, instance):
+        return isinstance(self.field, RelationshipField)
```

Comment:
Think this is backward -- should return `None` if it's not a `RelationshipField`